hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -365,21 +363,14 @@ public RegisteredCommand register(
     InetAddress dnAddress = Server.getRemoteIp();
     if (dnAddress != null) {
       // Mostly called inside an RPC, update ip
-      if (!useHostname) {
-        datanodeDetails.setHostName(dnAddress.getHostName());
-      }
+      datanodeDetails.setHostName(dnAddress.getHostName());
       datanodeDetails.setIpAddress(dnAddress.getHostAddress());
     }
 
-    String dnsName;
-    String networkLocation;
+    String dnsName = datanodeDetails.getIpAddress();
+    String hostName = datanodeDetails.getHostName();
```

Comment:
`dnsName` and `hostName` roughly mean the same thing.  Please use `ipAddress` for IP address.